### PR TITLE
Added persistent segtree explanations + problems

### PR DESCRIPTION
**Pull request by dolphingarlic**

### CODE REVIEW

File: content/6_Advanced/Persistent.mdx [hunk 1]
```diff
@@ -45,7 +45,21 @@ place (hence persistence).
 
 ### Solution
 
-<IncompleteSection />
+Since this problem involves range queries, we'll use some type of segment tree
+to solve it. (We can also use a Fenwick tree, but that's much harder to make
+persistent.)
+
+When dealing with problems involving multiple dimensions, it's often helpful to
+try view one of those dimensions as time. In this problem, we'll view the index
```
Comment (bqi343) [inline content/6_Advanced/Persistent.mdx:53]: > try view

File: content/6_Advanced/Persistent.mdx [hunk 2]
```diff
@@ -45,7 +45,21 @@ place (hence persistence).
...
+of each array as its time dimension.
+
+Using a persistent segment tree, we can then turn the problem into the
+following:
+
+- Type 1 queries involve a point update on the segment tree at some time.
+- Type 2 queries involve a range query on the segment tree at some time.
+- Type 3 queries involve copying the root of the segment tree at some time and
+  appending it to the array of segment tree roots at each time.
```
Comment (bqi343) [inline content/6_Advanced/Persistent.mdx:62]: > at each time.

I think you can remove this

File: content/6_Advanced/Persistent.mdx [hunk 3]
```diff
@@ -127,13 +141,34 @@ int main() {
...
 </LanguageSection>
 
-### Problems
+### Application 1 - Static 2D Range Sums on Large Grids
 
-A nice thing about persistent segment trees is that they can be used for online
-2D static range sum queries in $\mathcal{O}(\log N)$ time (think about it like
-prefix sums).
+Persistent segment trees can be used for online 2D static range sum queries in
+$\mathcal O(\log N)$ time (think of it like prefix sums.)
```
Comment (bqi343) [inline content/6_Advanced/Persistent.mdx:147]: period should go outside of parentheses

File: content/6_Advanced/Persistent.mdx [hunk 4]
```diff
@@ -127,13 +141,34 @@ int main() {
...
-<IncompleteSection />
+Note that 2D Fenwick trees with coordinate compression often also work for this
+(and are easier to implement), but it's still good to know this application.
+
+### Application 2 - Largest Interval Completely Inside a Range
+
+Consider the following problem:
+
+> Given $N$ intervals on the number line, answer $Q$ queries of the form "what
+> is the largest interval completely contained inside the range $[x, y]$?
```
Comment (bqi343) [inline content/6_Advanced/Persistent.mdx:157]: quotation doesn't end

File: content/6_Advanced/Persistent.mdx [hunk 5]
```diff
@@ -127,13 +141,34 @@ int main() {
...
+>
+> $N, Q \leq 10^5$.
+
+Since each interval has two dimensions (i.e. left and right endpoints $l_i$ and
+$r_i$), we can view it as a *point* on the number line at $l_i$ with "value"
+$r_i - l_i$ that was inserted at time $r_i$.
+
+Now, each query becomes "what is the most valuable point in the range $[x, y]$
+that was inserted at or before time $y$?" This is much easier to handle, so we
+can solve this problem in $\mathcal O(Q \log N)$ time.
```
Comment (bqi343) [inline content/6_Advanced/Persistent.mdx:167]: is there somewhere you can submit this?

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/Persistent.mdx (hunk 5 above)
Comment (dolphingarlic) [reply content/6_Advanced/Persistent.mdx:167]: I don't think so. @frodakcin @HarryWangATX can you guys try to make this a problem on the cf group please?

File: content/6_Advanced/Persistent.mdx [hunk 6]
```diff
@@ -50,16 +50,16 @@ to solve it. (We can also use a Fenwick tree, but that's much harder to make
 persistent.)
 
 When dealing with problems involving multiple dimensions, it's often helpful to
-try view one of those dimensions as time. In this problem, we'll view the index
-of each array as its time dimension.
+try view one of those dimensions as time. In this problem, we'll try view the
```
Comment (bqi343) [inline content/6_Advanced/Persistent.mdx:53]: shouldn't this say "try viewing" instead of "try view"?

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Comment by dolphingarlic** [comment]
Also what should go under the persistent array section?

File: content/6_Advanced/Persistent.mdx [hunk 7]
```diff
@@ -45,7 +45,21 @@ place (hence persistence).
 
 ### Solution
 
-<IncompleteSection />
+Since this problem involves range queries, we'll use some type of segment tree
+to solve it. (We can also use a Fenwick tree, but that's much harder to make
+persistent.)
+
+When dealing with problems involving multiple dimensions, it's often helpful to
+try view one of those dimensions as time. In this problem, we'll try viewing the
```
Comment (bqi343) [inline content/6_Advanced/Persistent.mdx:53]: > try view

you only fixed one occurrence

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Comment by bqi343** [comment]
> Also what should go under the persistent array section?

persistent array implementation (presumably it's easier to lead with that than a persistent segtree?)

File: content/6_Advanced/Persistent.mdx (hunk 7 above)
Comment (bqi343) [reply content/6_Advanced/Persistent.mdx:53]: sure

**Comment by dolphingarlic** [comment]
Do you mean a straightforward O(NM) implementation or a more efficient one like http://web.mit.edu/andersk/Public/6.851-presentation.pdf?

**Comment by bqi343** [comment]
> Do you mean a straightforward O(NM) implementation or a more efficient one like http://web.mit.edu/andersk/Public/6.851-presentation.pdf?

I meant log n per lookup and update

**Review by bqi343 — APPROVED** [review]

**Pull request merged**